tests/test_compiler.py

```diff
@@ -32,6 +32,31 @@
 # pylint: disable=too-few-public-methods, too-many-public-methods
 
 
+# TODO: Update this test for the code coverage after Catalyst 0.4.0 release
+def test_catalyst_incompatible():
+    """Test qjit with an incompatible Catalyst version < 0.4.0"""
+
+    qml.compiler.compiler.AvailableCompilers.names_versions["catalyst"] = False
+    dev = qml.device("lightning.qubit", wires=1)
+
+    @qml.qnode(dev)
+    def circuit():
+        qml.PauliX(0)
+        return qml.state()
+
+    with pytest.raises(
+        CompileError, match="PennyLane-Catalyst 0.4.0 or greater is required, but installed"
+    ):
+        qml.qjit(circuit)()
+    qml.compiler.compiler.AvailableCompilers.names_versions["catalyst"] = True
```

Comment:
This sort of set and reset is better handled via the `monkeypatch` fixture, check https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.setitem.

Reply:
Fixed [65a1efc](https://github.com/PennyLaneAI/pennylane/pull/4979/commits/65a1efc6f99dd1236bc133900139803eb832e197)